ci/tests/puppeteer/run.sh

```diff
@@ -235,13 +267,56 @@ if [[ "${REBUILD}" == "true" && "${RERUN}" != "true" ]]; then
   FLAGS=$(echo $BUILDFLAGS | sed 's/ //')
   printgreen "\nBuilding CAS found in $PWD for dependencies [${dependencies}] with flags [${FLAGS}]"
 
-  ./gradlew :webapp:cas-server-webapp-${project}:build \
-    -DskipNestedConfigMetadataGen=true -x check -x javadoc --build-cache --configure-on-demand --parallel \
-    ${BUILD_SCRIPT} ${DAEMON} -DcasModules="${dependencies}" -q ${BUILDFLAGS}
+  BUILD_CACHE=""
+  if [[ "$GRADLE_BUILDCACHE_PSW" != "" || "$CI" != "true" ]]; then
```

Comment:
You should always include the build cache flag I think. You'd be in read-only mode if the PSW is not defined, which is what you'd want anyway.